icepyx/core/orders.py

```diff
@@ -0,0 +1,201 @@
+from pathlib import Path
+from typing import Any, Dict, Union
+
+import earthaccess
+
+
+class DataOrder:
+    """
+    A class representing an order for Harmony data processing.
+
+    Attributes
+    ----------
+    job_id : str
+        The ID of the Harmony job.
+    type : str
+        The type of order (e.g., "subset").
+    granules : list
+        A list of granules included in the order.
+    harmony_client : object
+        The Harmony API client used for interacting with the service.
+
+    Methods
+    -------
+    id()
+        Returns the job ID of the order.
+    resume()
+        Resumes the order if it has been paused due to exceeding granule limits.
+    pause()
+        Pauses the order.
+    status()
+        Retrieves the status of the order.
+    download_granules(path, overwrite=False)
+        Downloads the granules associated with the order.
+    download(path, overwrite=False)
+        Downloads the granules, waiting until they are ready if necessary.
+    """
+
+    HARMONY_BASE_URL = "https://harmony.earthdata.nasa.gov/workflow-ui/"
+
+    def __init__(self, job_id, type, granules, harmony_client):
+        """
+        Initialize a DataOrder object.
+
+        Parameters
+        ----------
+        job_id : str
+            The ID of the Harmony job.
+        type : str
+            The type of order (e.g., "subset").
+        granules : list
+            A list of granules included in the order.
+        harmony_client : object
+            The Harmony API client.
+        """
+        self.job_id = job_id
+        self.harmony_api = harmony_client
+        self.granules = granules
+        self.type = type
+
+    def __str__(self) -> str:
+        return f"DataOrder(job_id={self.job_id}, type={self.type}, granules={self.granules})"
+
+    def _repr_html_(self) -> str:
+        # Create a link using the <a> tag
+        status = self.status()
+        link_html = f'<a target="_blank" href="{self.HARMONY_BASE_URL}{self.job_id}">View Details</a>'
+        # Create a self-contained HTML table with a single row
+        html = f"""
+        <table border="1">
+            <thead>
+                <tr>
+                    <th>Job ID</th>
+                    <th>Type</th>
+                    <th>Status</th>
+                    <th>Details</th>
+                </tr>
+            </thead>
+            <tbody>
+                <tr>
+                    <td>{self.job_id}</td>
+                    <td>{self.type}</td>
+                    <td>{status["status"]}</td>
+                    <td>{link_html}</td>
+                </tr>
+            </tbody>
+        </table>
+        """
+        return html
+
+    def __repr__(self) -> str:
+        return self.__str__()
+
+    def id(self) -> str:
+        """
+        Get the job ID of the order.
```

Comment:
Would prefer if we just use `job_id`, because there is a python built-in function called [`id`](https://docs.python.org/3/library/functions.html#id), and want to avoid confusion.

```suggestion
    def job_id(self) -> str:
        """
        Get the job ID of the order.
```